src/main.rs

```diff
@@ -1,3 +1,95 @@
+use rand::{rng, Rng};
+use std::cmp::Ordering;
+use std::io;
+
+const GENERATED_NUMBER_UPPER_LIMIT: i32 = 100;
+const GENERATED_NUMBER_LOWER_LIMIT: i32 = 1;
+
 fn main() {
-    println!("Hello, world!");
+    let generated_number: i32 = initialize_guessing_game();
+
+    run_guessing_game_loop(generated_number);
+}
+
+/// This function prints the game instructions to the user
+/// And randomly generates a number between 1-100 for the user to guess
+///
+/// # Returns
+///
+/// The generated number for the user to guess
+fn initialize_guessing_game() -> i32 {
+    println!(
+        "Welcome To Guess The Number!\n You Should enter a number between {}-{}",
+        GENERATED_NUMBER_LOWER_LIMIT, GENERATED_NUMBER_UPPER_LIMIT
+    );
+
+    let mut rng = rng();
+    let generated_number =
+        rng.random_range(GENERATED_NUMBER_LOWER_LIMIT..=GENERATED_NUMBER_UPPER_LIMIT);
+
+    return generated_number;
```

Comment:
This could be unionized, rendering the variables useless.

```rust
{
    rng()
        .random_range(GENERATED_NUMBER_LOWER_LIMIT..=GENERATED_NUMBER_UPPER_LIMIT)
}
```